### PR TITLE
Remove phone number validation on the client side

### PR DESCRIPTION
**Pull request by samtstern**
See #812 

@ashwinraghav is this the intended behavior? The libphonenumber call is only made for SmartLock hints but not for manually entered phone numbers.

### CODE REVIEW

**Comment by samtstern** [comment]
cc: @isaisachen

**Comment by ashwinraghav** [comment]
;tldr
It was intentional, but the wrong decision in hindsight.

Here was how we thought of this:
1) Smartlock hints needed it (at the time) since the phone number formats returned were not in the E164 format. We started using telephony's normalization (assuming it would somehow be updated regularly enough since libPhoneNumber that backs it changes every week).
2) We piggybacked on the phone number normalization for manually entered phone numbers to save on service calls for fat-fingered phone numbers.

To state the obvious, both 1 and 2 are yielding false negatives since the lib is not updated as often as we would like. I would be ok getting rid of this validation for both 1 and 2 or just 2.

File: auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneNumberUtils.java [hunk 1]
```diff
@@ -93,6 +88,14 @@ static String formatPhoneNumber(@NonNull String phoneNumber, @NonNull CountryInf
     static String formatPhoneNumberUsingCurrentCountry(
             @NonNull String phoneNumber, Context context) {
         final CountryInfo currentCountry = PhoneNumberUtils.getCurrentCountryInfo(context);
+
+        // TODO(samstern): Remove this call once the next version of Play servics is released
+        //                 estimated timelime August 10th
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```
Comment (ashwinraghav) [inline auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneNumberUtils.java]: Might be worth adding a (known bug) to the commend that hint selector is still susceptible to the false negatives of this method.

**Comment by ashwinraghav** [comment]
LGTM

**Pull request merged**